modules/miscutil/lib/orcidutils.py

```diff
@@ -416,18 +416,84 @@ def _orcid_fetch_works(pid):
                                                     existing_papers, doid):
 
-        xml_to_push = OrcidXmlExporter.export(orcid_small_list, 'works.xml')
-        xml_to_push = RE_ALLOWED_XML_1_0_CHARS.sub('',
-                                                   xml_to_push).encode('utf-8')
-        yield (doid, xml_to_push, len(orcid_small_list))
+        yield (doid, orcid_small_list, len(orcid_small_list))
+
+
+def _push_few_works(doid, short_list, number, pid, token):
+    '''Push works with fallback in case of duplicates.
+
+    If there is a duplicate, the function will remove the problematic
+    paper from the list and repush.
+
+    @return: did the push finish successfully
+    @rtype: bool
+    '''
+    url = CFG_OAUTH2_CONFIGURATIONS['orcid']['member_url'] + \
+        doid + '/orcid-works'
+
+    xml_to_push = OrcidXmlExporter.export(short_list, 'works.xml')
+    xml_to_push = RE_ALLOWED_XML_1_0_CHARS.sub('',
+                                               xml_to_push).encode('utf-8')
+
+    bibtask.write_message(
+        "Pushing %s works for %s. Personid: %s" %
+        (number, doid, pid))
+
+    headers = {'Accept': 'application/vnd.orcid+xml',
+               'Content-Type': 'application/vnd.orcid+xml',
+               'Authorization': 'Bearer ' + token
+               }
+
+    response = requests.post(url, xml_to_push, headers=headers)
+
+    code = response.status_code
+
+    if code == 401:
+        # The token has expired or the user revoke his token
+        delete_token(pid)
+        bibtask.write_message("Token deleted for %s" % (pid,))
+        register_exception(subject="The ORCID token expired.")
+        return True
+    elif code != 201 and code != 200:
+        try:
+            response.raise_for_status()
+        except HTTPError, exc:
+            m = re.search(r"have the same external id \"(.*)\"", response.text)
+            if m:
+                bibtask.write_message("Works have the same doi: %s" % m.groups()[0])
+                wrong_id = m.groups()[0]
+                shorter_list = [x for x in short_list if
+                                all(t[1] != wrong_id for
+                                    t in x['work_external_identifiers'])]
+                blacklist = {}
+                try:
+                    blacklist = json.loads(open(CFG_ORCIDUTILS_BLACKLIST_FILE,
+                                                'r').read())
+                except IOError:
+                    # No such file
+                    pass
+                if blacklist:
+                    if doid in blacklist:
+                        blacklist[doid].append(wrong_id)
+                    else:
+                        blacklist[doid] = [wrong_id]
+                    json.dump(blacklist, open(CFG_ORCIDUTILS_BLACKLIST_FILE,
+                                              'w'))
+                return _push_few_works(doid, shorter_list, number, pid, token)
```

Comment:
In case all ements are duplicate and `shorter_list` is empty you should not call again `_push_few_works()`. Otherwise ORCID will probably fail...